### PR TITLE
Gym rats grow in size the stronger they are, up to 2x their original size at max bulk.

### PR DESCRIPTION
**Pull request by Eneocho**
## What this does
Makes it so whenever a gym rat, pompadour rat or ROID RAT eats meat, they grow in size, up to 2x their original size at MAX BULK. When exposed to soymilk, they accordingly become smaller and smaller.
When feeding on meat:
ezgif-3-f8bb72626e
When sprayed with soymilk:
ezgif-3-d0bae15303
## Why it's good
Gives a way to check just how strong a gym rat is, the bigger they are, the stronger. 
:cl:
 * rscadd: Gym Rats, Pompadour Rats and Roid Rats become bigger when bulking, maxing out at 2x their original size at max bulk. When exposed to soymilk, which reduces their bulk, they go smaller accordingly. They can't go smaller than their original size.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/gym_rats.dm [hunk 1]
```diff
@@ -232,6 +242,17 @@
 			if(maxHealth >= 20)
 				visible_message("<span class='warning'>[src] seems to shrink as the soymilk washes over them! Its muscles look less visible...</span>")
 				maxHealth-=10
+				if(initial(maxhealth) = 30) //regular gym rat
```
Comment (twirling-jesters) [inline code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: maxhealth should be maxHealth here

Comment (twirling-jesters) [reply code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: ah nevermind you got it

File: code/modules/mob/living/simple_animal/hostile/gym_rats.dm [hunk 2]
```diff
@@ -99,6 +101,15 @@
 		playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
 		if(maxHealth < health_cap) // Are we below our max gainz level? Add on some max hp!
 			adjust_hp(5)
+			if(rattype = regular) //regular gym rat
```
Comment (twirling-jesters) [inline code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: rattype = regular -> rattype == "regular"

File: code/modules/mob/living/simple_animal/hostile/gym_rats.dm [hunk 3]
```diff
@@ -232,6 +243,17 @@
 			if(maxHealth >= 20)
 				visible_message("<span class='warning'>[src] seems to shrink as the soymilk washes over them! Its muscles look less visible...</span>")
 				maxHealth-=10
+				if(rattype = regular) //regular gym rat
```
Comment (twirling-jesters) [inline code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: same as above, replace with rattype == "regular"

File: code/modules/mob/living/simple_animal/hostile/gym_rats.dm [hunk 4]
```diff
@@ -232,6 +243,17 @@
...
+					SCALERATE -= (1/7)
+				else if(rattype = pompadour) //pompadour
```
Comment (twirling-jesters) [inline code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: same as above, use == and "pompadour" etc.

File: code/modules/mob/living/simple_animal/hostile/gym_rats.dm (hunk 2 above)
Comment (Eneocho) [reply code/modules/mob/living/simple_animal/hostile/gym_rats.dm]: ah that makes sense, forgot about double trues

**Comment by tinysaturn** [comment]
NOOOOOOOOO MY HARD TO CLICK ON SPRITE NOOOOOOOOO
HOW WILL I SOLO WRECK GYGAXES IN HAND TO HAND COMBAT NOOOOOOOOOOOOW

**Comment by Exxion** [comment]
Typically, all-caps names indicate consts or defines. `SCALERATE` should be renamed to something not all-caps. `scalerate`, `scale_rate`, or `scaleRate`, for example

**Comment by Eneocho** [comment]
> Typically, all-caps names indicate consts or defines. `SCALERATE` should be renamed to something not all-caps. `scalerate`, `scale_rate`, or `scaleRate`, for example

I'll change it then, I had used SCALERATE on my testing file.

**Comment by Optimism333** [comment]
You should share an image/preview of how it looks.

**Comment by Eneocho** [comment]
> You should share an image/preview of how it looks.

Added a gif for going buff and one for being soymilked.

**Comment by twirling-jesters** [comment]
Any way you could do a small translate up every time the mouse is scaled, just so the sprite stays centered on the tile? As it stands, the mouse starts bleeding into the tile below it as it grows. Something like

M.Translate(0, scalerate)

might do the trick

**Comment by Eneocho** [comment]
> Any way you could do a small translate up every time the mouse is scaled, just so the sprite stays centered on the tile? As it stands, the mouse starts bleeding into the tile below it as it grows. Something like
> 
> M.Translate(0, scalerate)
> 
> might do the trick

I'll look into it, but I'll have to admit that I've never seen M.translate before, might take a while.

**Comment by Eneocho** [comment]
After some testing the ideal translate value seems to be 9 (8 from translating + default 1). Implemented. A bit jury rigged, but it works.

**Comment by Killette2** [comment]
Since this might be the only time I get to nitpick, could we consider changing the scaling to 1.5x instead of 2? I watched the gifs in the OP and it seems around 14 is full size, and 7 bites is the in between. The size at 7 bites (which I assume 1.5x scaling would be) is more than enough to address the hard to click issue the mob benefits from, but I think at 2x the scaling is a little too intense

**Comment by Eneocho** [comment]
I originally intended to have it at 1.5x but it looked all deformed since pixels needed to be stretched to fit. At 2x this doesn't happen, it looks better and very obviously displays that the rats are a threat now. Sure pixel stretching still happens while getting to 2x, but at max bulk (which is very quickly reached since people tend to also make meat for the rats) it looks fine.

**Comment by Hinaichigo** [comment]
I like how the original rats are slightly larger than normal rats and it makes it funny when they stand up. Making them this big seems sorta like overdoing it. Plus, I don't like pixel stretching. Maybe some overlays for  slightly bigger muscles could look nice.

**Comment by Optimism333** [comment]
I like the idea, but I do think they are a little too big as well.

**Pull request merged**